pq-crypto/s2n_kyber_512_evp.h

```diff
@@ -0,0 +1,28 @@
+/*
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#pragma once
+
+#include <openssl/evp.h>
+
+#include "tls/s2n_kem.h"
+
+#if (defined(S2N_AWSLC_KYBER) && defined(OPENSSL_IS_AWSLC) && AWSLC_API_VERSION >= 20)
+#define AWS_LC_SUPPORTS_KYBER_512
+#endif
+
+int s2n_kyber_512_evp_generate_keypair(OUT unsigned char *public_key, OUT unsigned char *private_key);
```

Comment:
Is this our chance to move from `unsigned char` to `uint8_t`?

Reply:
I'd be happy to take a stab at it, anything I should be mindful of?

Reply:
Not sure how it all fit together. I believe the first iteration used `unsigned char` because the API fronting the imported algorithm implementations exposed a function signatures with that type.

AWS-LC prefers `uint8_t` type though. So, maybe we can align that. I also think `uint8_t` is marginally better.

Reply:
Went ahead and made the change and everything is compiling/testing fine.